### PR TITLE
Further cleanup in File classes

### LINKED ISSUE #1662: Clean up File, FileUnix and FileWin32 classes
For example, rename `rm` and `rmrf` methods, they're ugly.
There is too much indentation in the latter classes, and a lot of methods we don't use, and a lot that can be moved to properties (or shouldn't be both method and property).
There's also a big link to a commit on upstream rock.

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1662

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/system/io/File.ooc [hunk 1]
```diff
@@ -165,13 +165,17 @@ File: abstract class {
 	getReducedFile: func -> This { This new(this getReducedPath()) }
 	getChildrenNames: abstract func -> VectorList<String>
 	getChildren: abstract func -> VectorList<This>
-	rm: func -> Bool { _remove(this) }
-	rmrf: func -> Bool {
-		if (this dir())
-			for (child in this getChildren())
-				if (!child rmrf())
+	remove: func -> Bool { _remove(this) }
+	removeForceRecursive: func -> Bool {
```
Comment (sebastianbaginski) [inline source/system/io/File.ooc]: Maybe just `removeRecursive` ?

Comment (marcusnaslund) [inline source/system/io/File.ooc]: You're right, there's nothing `-f` about this

**Comment by marcusnaslund** [comment]
Updated.

**Comment by marcusnaslund** [comment]
Changed my mind, I want to rename some more stuff, like `mkdir`, I will take this off review for now.

**Comment by marcusnaslund** [comment]
Alright, back to review we go. @sebastianbaginski ?

**Comment by sebastianbaginski** [comment]
Great.

**Pull request merged**